lib/charms/rolling_ops/v0/rollingops.py

```diff
@@ -0,0 +1,353 @@
+"""A library to enable charms to implement "rolling" or "serialized" operations. E.g., a
+rolling restart.
+
+You may use this library directly, or extend it to customize behavior.
+
+For example, in order to implement a rolling restart, a charm author would need to add the
+following lines of code, to the following files (note the consistent use of the name "restart")
+to namespace this particular rolling op:
+
+Add a peer relation to `metadata.yaml`:
+```yaml
+peers:
+    restart:
+        interface: rolling_op
+```
+
+To enable a human operator to trigger the restat, add an action to `actions.yaml`:
+```yaml
+restart:
+  description: Restarts a service
+  params: {}
+```
+
+Import the library, and enable it by doing the following:
+
+1. Override the Charm's `on` property with the RollingEvents class from the library.
+2. Add a method containing code to be executed to the Charm object. Here, we call this
+method `_restart`.
+3. Initialize a RollingOpsManager class, passing in the Charm, name of the peer relation,
+restart handler, and action.
+
+src/charm.py
+```python
+# ...
+from charms.rolling_ops.v0.rollingops import RollingOpsManager, RollingEvents
+# ...
+
+class SomeCharm(...):
+    # ...
+    on = RollingEvents
+    # ...
+
+    def __init__(...)
+        # ...
+        self.restart = RollingOpsManager(self, 'restart', self._restart, self.on.restart_action)
```

Comment:
I wonder if we should make the args kwarg only here? I feel like it'll improve the readability of charms quite a bit?

Reply:
We could. I initially had coded this as keyworded args, and got rid of them for brevity's sake.

I really would like to chop out at least two more of the args, but I'm not 100% certain how to do so.

(It's possible that the way we resolved multiple sets of events from multiple libraries will also lead us to namespacing them, which could eliminate the 'restart' arg.)